### PR TITLE
feat(microservice): hybrid microservice can inherit config

### PR DESCRIPTION
**Pull request by 2Kable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #4261 

## What is the new behavior?
```
app.connectMicroservice(transportOptions, { inheritAppConfig: true });
```
When providing `inheritAppConfig` the hybrid microservice will inherit the global configuration.
allowing use of global interceptor and such

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by 2Kable** [comment]
I started working on the documentation update

File: packages/common/interfaces/microservices/nest-hybrid-options.interface.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export interface NestHybridOptions {
```
Comment (kamilmysliwiec) [inline packages/common/interfaces/microservices/nest-hybrid-options.interface.ts]:
```suggestion
export interface NestHybridApplicationOptions {
```

Comment (kamilmysliwiec) [reply packages/common/interfaces/microservices/nest-hybrid-options.interface.ts]: and this change would require a few changes in other locations too (filename, method arg name etc)

**Comment by kamilmysliwiec** [comment]
Thanks for your contribution @2Kable!

**Comment by 2Kable** [comment]
Interface has been changed
docs has been added: https://github.com/nestjs/docs.nestjs.com/pull/1100

**Comment by kamilmysliwiec** [comment]
Thanks @2Kable :)

**Pull request merged**